src/sentry/static/sentry/app/views/settings/components/forms/apiForm.tsx

```diff
@@ -1,41 +1,40 @@
-import PropTypes from 'prop-types';
 import React from 'react';
 
 import {Client} from 'app/api';
 import {addLoadingMessage, clearIndicators} from 'app/actionCreators/indicator';
 import {t} from 'app/locale';
 import Form from 'app/views/settings/components/forms/form';
 
-export default class ApiForm extends React.Component {
-  static propTypes = {
-    ...Form.propTypes,
-    onSubmit: PropTypes.func,
-    apiMethod: PropTypes.string.isRequired,
-    apiEndpoint: PropTypes.string.isRequired,
-  };
-
-  constructor(props, context) {
-    super(props, context);
-    this.api = new Client();
-  }
+type Props = Form['props'] & {
+  onSubmit?: (data: Record<string, any>) => void;
```

Comment:
we usually order the props with the first ones not optional and the last ones optional 😉